### ISSUE TITLE
WooCommerce default pagination limit to 10 prevents import of all products

### ISSUE THREAD

**Issue by albertolopezma** [techreceptives/connector-woocommerce#6]
Reviewing the connector I found some issues in the import logic of products.

When importing products, no more than 10 results are included, this seems to be related to default pagination limit of 10 results per page in WooCommerce and default page=1 setting.

Using Python example code from WooCommerce module version 1.0.4, it is necessary to change the filter setting as in ` products?filter[limit]=9999` to get more than 10 products in an API call:

```python
from woocommerce import API

wcapi = API(
    url="http://example.com",
    consumer_key="ck_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    consumer_secret="cs_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    version="v2"
)

r = wcapi.get('products/list')
r.json()

s = wcapi.get('products/list?filter[limit]=9999')
s.json()

```

This seems to be the case for connector-woocommerce module. I think the connector uses the filter feature to filter by date, but leaves limit and pagination settings to their default values.

It may be necessary to use the filter limit parameter or, even better, to break down product and category import in batches of a fixed number of elements, i.e. 100 per run, running multiple times until every product is fetched.

**Comment by albertolopezma** [comment]
Reviewing the WooCommerce Woo Odoo plugin code, it looks like the `get_products_list()` function in `connector-woo-odoo/class-wc-api-products-list.php` for the `products/list` endpoint is based on the WooCommerce `get_products()` function in file `woocommerce/includes/api/class-wc-api-products.php`, which introduces pagination explicitly.

The `get_product_list()` function for the `products/list` endpoint could be based on the `get_product_count()` function in WooCommerce, which makes no use of pagination filtering and may work as the `products/category/list` endpoint does.

**Comment by parthivgls** [comment]
fixed with commit 954d078aafeabdd859ee91743632590f7abd60b8
